src/bp-core/admin/settings/bp-admin-setting-activity.php

```diff
@@ -68,7 +68,9 @@ public function settings_save() {
 
 		$bp->active_components = $active_components;
 		bp_core_install( $bp->active_components );
-		bp_core_add_page_mappings( $bp->active_components );
+
+		// Mapping the component pages in page settings except registration pages.
+		bp_core_add_page_mappings( $bp->active_components, 'keep', false );
```

Comment:
@surajkrsingh Can you check how many time this function call in the product?  I think page creation logic should call only once. 
Look like this is not the correct way. Ping me we will discuss it on call.

Reply:
@dipskakadiya 
This is called around 13 places in the product(on a different case and different functions) except for test cases.
Total 21 times including test cases.